### PR TITLE
TS-4737: Remove XML log configuration.

### PR DESCRIPTION
**Pull request by jpeach**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/568/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/464/ for details.

File: doc/admin-guide/files/records.config.en.rst [hunk 1]
```diff
@@ -2651,7 +2651,7 @@ Logging Configuration
    ===== ======================================================================
 
    For information on sending XML-based custom formats to the collation server,
```
Comment (shukitchan) [inline doc/admin-guide/files/records.config.en.rst]: get rid of "XML-based" here? :)

**Comment by zwoop** [comment]
Doing a quick git grep, I a few places that might need more cleanup?

proxy/logging/Log.h:    SEND_NON_XML_CUSTOM_FMTS,
proxy/logging/Log.h:    SEND_STD_AND_NON_XML_CUSTOM_FMTS,
proxy/logging/LogFilter.cc:LogFilterString::display_as_XML(FILE *fd)
proxy/logging/LogFilter.cc:LogFilterInt::display_as_XML(FILE *fd)
proxy/logging/LogFilter.cc:LogFilterIP::display_as_XML(FILE *fd)
proxy/logging/LogFilter.cc:LogFilterList::display_as_XML(FILE *fd)
proxy/logging/LogFilter.cc:    f->display_as_XML(fd);
proxy/logging/LogFilter.h:  virtual void display_as_XML(FILE *fd = stdout) = 0;
proxy/logging/LogFilter.h:  void display_as_XML(FILE *fd = stdout);
proxy/logging/LogFilter.h:  void display_as_XML(FILE *fd = stdout);
proxy/logging/LogFilter.h:  void display_as_XML(FILE *fd = stdout);
proxy/logging/LogFilter.h:  void display_as_XML(FILE *fd = stdout);
proxy/logging/LogFormat.cc:LogFormat::displayAsXML(FILE *fd)
proxy/logging/LogFormat.h:  void displayAsXML(FILE *fd = stdout);
proxy/logging/LogObject.cc:LogObject::displayAsXML(FILE *fd, bool extended)
proxy/logging/LogObject.cc:    m_format->displayAsXML(fd);
proxy/logging/LogObject.cc:    m_filter_list.display_as_XML(fd);
proxy/logging/LogObject.h:  void displayAsXML(FILE *fd = stdout, bool extended = false);

**Comment by zwoop** [comment]
Also, there are several section in docs the refer to the XML format. I'm ok with putting that on a different Jira issue though, and we can rope in @jsime for help (lets just not forget it).

**Comment by jpeach** [comment]
FWIW I'm leaving the docs in place for now and assuming that @jsime will take those changes on.

**Comment by jpeach** [comment]
I removed the XML debug routines, but I didn't touch the ``Log::CollationMode`` enum (at least for now).

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/474/ for details.

File: doc/admin-guide/files/records.config.en.rst (hunk 1 above)
Comment (jpeach) [inline doc/admin-guide/files/records.config.en.rst]: FWIW the main docs jira is [TS-4738](https://issues.apache.org/jira/browse/TS-4738).

**Pull request merged**